\documentclass[11pt]{report}
\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{caption}
\usepackage{subcaption}
\usepackage{amsmath}
\usepackage{afterpage}
\usepackage{pdflscape}
\usepackage{adjustbox}
\DeclareMathOperator*{\argmax}{argmax}
\DeclareMathOperator*{\argmin}{argmin}
\graphicspath{{images/}} 
\usepackage[ruled,vlined]{algorithm2e}
\usepackage{listings}
\usepackage{xcolor}
\usepackage{setspace}
\definecolor{codegreen}{rgb}{0,0.6,0}
\definecolor{codegray}{rgb}{0.5,0.5,0.5}
\definecolor{codepurple}{rgb}{0.58,0,0.82}
\definecolor{backcolour}{rgb}{0.95,0.95,0.92}
\lstdefinestyle{mystyle}{
    language=c,
    %basicstyle=\small\ttfamily,
    basicstyle=\linespread{0.8}\small\ttfamily,
    stepnumber=1,
    commentstyle=\color{codegreen},
    keywordstyle=\color{magenta},
    numberstyle=\tiny\color{codegray},
    stringstyle=\color{codepurple},
    numbersep=8pt,
    showstringspaces=false,
    breaklines=true,
    frame=top,
    frame=bottom,
    aboveskip=1.5em,
    belowskip=1.5em,
    captionpos=b
}

\def\frontmatter{%
    \pagenumbering{roman}
    \setcounter{page}{1}
    \renewcommand{\thesection}{\Roman{section}}
}%

\def\mainmatter{%
    \pagenumbering{arabic}
    \setcounter{page}{1}
    \setcounter{section}{0}
    \renewcommand{\thesection}{\arabic{section}}
}%

\def\backmatter{%
    \setcounter{section}{0}
    \renewcommand{\thesection}{\Alph{section}}
}%

\lstset{style=mystyle}
\newcommand{\changefont}{%
    \fontsize{11}{11}\selectfont
}
\usepackage{ragged2e}
\newenvironment{abbreviations}{\begin{list}{}{\renewcommand{\makelabel}{\abbrlabel}}}{\end{list}}
\usepackage{changepage}
\setlength{\RaggedRightParindent}{\parindent}
\usepackage[a4paper,margin=1in,bindingoffset=6mm]{geometry}
\usepackage{fancyhdr}
\pagestyle{fancy}
\newcommand{\abbrlabel}[1]{\makebox[3cm][l]{\textbf{#1}\ \dotfill}}

\usepackage{setspace} 
\fancyhf{}
\fancyhead[LE,RO]{\changefont \slshape \rightmark} %section
\fancyhead[RE,LO]{\changefont \slshape \leftmark} %chapter
\fancyfoot[C]{\changefont \thepage} %footer
\doublespacing
\RaggedRight 


\usepackage[backend=biber,refsection=chapter,style=ieee,maxbibnames=99,citestyle=numeric-comp]{biblatex}
\addbibresource{references.bib}

\begin{document}


\begin{titlepage}
\newgeometry{bottom=.5in}
   \begin{singlespacing}
   \begin{center}
       \vspace*{1in}
       \textsc{IDENTIFICATION OF FUNCTIONAL DOMAINS IN NON-CODING RNA}\\
       \vspace{1in}
       Daniel Sprague\\
       \vspace{1in}
       A dissertation submitted to the faculty at the University of North Carolina at Chapel Hill in partial fulfillment of the requirements for the degree of Doctor of Philosophy in the Department of Pharmacology in the School of Medicine.\\
       \vspace{1in}
       Chapel Hill\\
       2020
    \end{center}
    \end{singlespacing}
    \vspace{1in}
    \begin{flushleft}
    \hspace{3.9in} Approved by:\\
     \hspace{3.9in}  J. Mauro Calabrese\\
     \hspace{3.9in}   Lee Graves\\
      \hspace{3.9in}  Daniel Dominguez\\
      \hspace{3.9in}  Jeremy Purvis\\
      \hspace{3.9in}   Daniel Schrider
      \end{flushleft}
\newgeometry{bottom=1in}
\end{titlepage}
\frontmatter

\clearpage 
\thispagestyle{plain}
\begin{center}
    \null 
    \vfill
    \begin{singlespacing}
    \textcopyright \ 2020\\
    Daniel Sprague\\
    \textsc{All rights reserved}
    \end{singlespacing}
\end{center}

\renewcommand\abstractname{\normalfont\textsc{Abstract}}

\begin{abstract}
\stepcounter{page}
\thispagestyle{plain}
\begin{center}
Daniel Sprague: Identification of functional domains in non-coding RNA\\
(Under the direction of J. Mauro Calabrese)
\end{center}
Long non-coding RNAs (lncRNAs) are known to be important regulators of gene expression and other cellular functions. However, only a very small proportion of lncRNAs have been extensively studied. The remainder exist largely as annotations in a database with no known function, if any. A primary challenge to understanding how lncRNAs function is the poorly understood sequence-to-function relationship relative to protein coding genes. Within lncRNA transcripts, boundaries of functional sequence are not explicitly defined by exon-intron boundaries, and the code by which lncRNAs derive function is not nearly as explicit as in a protein coding reading frame. To address these challenges, we have developed a probabilistic framework, \emph{hmmSEEKR}, for identifying where within a non-coding RNA functional regions may be located based off of enrichment of short motifs, or $k$-mers. 

We used \emph{hmmSEEKR} to identify functional sequence domains in several lncRNAs that silence gene expression through recruitment of Polycomb, using \emph{XIST} as a model transcript. These predicted sequence domains share no detectable linear sequence alignment with \emph{XIST}; however, they share high $k$-mer based similarity with known functional domains in \emph{XIST} and precisely coincided with the location of RNA binding protein (RBP) interactions known to be important for Polycomb mediated silencing. Furthermore, we were able to extend our analysis to the entire transcriptome and identify many \emph{XIST}-like sequence domains throughout the transcriptome that interact with Polycomb-associated RBPs. We have packaged these algorithms into python-based software and have included an in-depth walk-through of the code and tutorial of how to analyze sequences using \textit{hmmSEEKR. } 
\end{abstract}
\chapter*{\center Acknowledgements}
\stepcounter{page}
\begin{center}
This work wouldn't have been possible without all the fantastic people who helped me through the end. I would like to thank my advisor, Mauro Calabrese, for always pushing me to improve as well as for the opportunity to work on some very exciting science. I would also like to thank all the friends and family who made this work possible. 
\end{center}
\tableofcontents
\listoffigures
\listoftables
\clearpage
\chapter*{List of Abbreviations}
\begin{abbreviations}
\item [$\in$] in
\item [$\mapsto$] maps to
\item [$\sum_X$] sum the values over all items in the set $X$
\item [``+"] query state in \emph{hmmSEEKR}
\item [``-"] null state in \textit{hmmSEEKR}
\item [bp] Basepair
\item [BLAST] Basic local alignment search tool

\item [eCLIP] Enhanced cross-linking immunoprecipitation 
\item [HMM] Hidden Markov Model
\item [HMMER] Profile HMM tool for sequence alignment 
\item[hmmSEEKR] Hidden Markov Model based Sequence Evaluation through $k$-mer representation
\item [KL-Divergence] Kullback-Leibler Divergence
\item [KR(1-4)] Koala \textit{Rsx} Repeats 1,2,3,4
\item [lncRNA] Long non-coding RNA
\item [$P(X,Y)$] Joint probability distribution
\item [$P(X|Y)$] Conditional probability distribution
\item [MX(A-E)] Mouse \textit{Xist} Repeats A,B,C,E
\item [HX(A-E)] Human \textit{XIST} Repeats A,B,D,E
\item [mRNA] Messenger RNA
\item [OR(1-4)] Opossum \textit{Rsx} Repeats 1,2,3,4

\item [PRC] Polycomb Repressive Complex
\item [PWM] Position weight matrix 

\item [RBP] RNA Binding Protein
\item [RNA-IP] RNA immunoprecipitation
\item [RNA-seq] RNA sequencing 
\item [SAF] Simple annotation format
\item [SEEKR] Sequence evaluation through $k$-mer representation
\item [XCI] X chromosome inactivation
\item [Xi] Inactive X
\item [\textit{XIST}] Human X inactivation specific transcript
\item [\textit{Xist}] Mouse X inactivation specific transcript

\end{abbreviations}
\clearpage



\mainmatter

\chapter{Introduction}
\input{chapters/introduction}
\chapter{Xist and Rsx}
\input{chapters/chapter02}
\chapter{hmmSEEKR}
\input{chapters/chapter03}
\chapter{SEEKR $k$-mers}
\input{chapters/chapter04}
\chapter{Conclusion}
\input{chapters/conclusion}


\end{document}
